esmvalcore/_recipe/recipe.py

```diff
@@ -246,7 +245,7 @@ def _get_default_settings(dataset: Dataset) -> PreprocessorSettings:
 
     settings = {}
 
-    if _derive_needed(dataset):
+    if dataset._derivation_necessary():  # noqa: SLF001 (will be replaced soon)
```

Comment:
your noqa is being ignored, I think it's ignoring it because you added inline text, Manu

Reply:
Yeah, it's not ignored, otherwise pre-commit would raise an error